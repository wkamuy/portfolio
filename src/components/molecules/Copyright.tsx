import { FC } from 'react';
import Typography from '@material-ui/core/Typography';

type CopyrightProps = {
  authorName: string;
};

const Copyright: FC<CopyrightProps> = ({ authorName }) => (
  <Typography variant="body2" color="textSecondary" align="center">
    {'Â© '}
    {new Date().getFullYear()} {authorName}.
  </Typography>
);

export default Copyright;
