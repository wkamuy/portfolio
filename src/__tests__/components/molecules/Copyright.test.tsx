import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Copyright from '../../../components/molecules/Copyright';

describe('<Copyright />', () => {
  it('should render correctly', () => {
    const mockDate = new Date(2021, 1, 1);
    const spy = jest
      .spyOn(global, 'Date')
      .mockImplementation(() => (mockDate as unknown) as string);

    const tree = renderer.create(<Copyright authorName="wkamuy" />).toJSON();
    expect(tree).toMatchSnapshot();

    spy.mockRestore();
  });
});
