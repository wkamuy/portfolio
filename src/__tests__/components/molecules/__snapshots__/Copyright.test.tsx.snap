// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Copyright /> should render correctly 1`] = `
<p
  className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
>
  Â© 
  2021
   
  wkamuy
  .
</p>
`;
